src/tsc/tsc.ts

```diff
@@ -322,6 +322,10 @@ namespace ts {
         const compilerOptions = program.getCompilerOptions();
         if (compilerOptions.diagnostics || compilerOptions.extendedDiagnostics) {
             statistics = [];
+
+            // @ts-ignore
+            if (typeof gc === "function") { gc(); }

```

Comment:
This isn't necessary. We explicitly call `gc()` before collecting memory usage in `sys.getMemoryUsage` when `--expose-gc` is used:

https://github.com/microsoft/TypeScript/blob/6f079a4ebcb78489eabb523faaf1b69f88e80bd7/src/compiler/sys.ts#L1212-L1217